tools/src/main/java/com/nedap/archie/flattener/FullArchetypeRepository.java

```diff
@@ -53,17 +56,55 @@ default void compile(MetaModels models) {
      */
     default ValidationResult compileAndRetrieveValidationResult(String archetypeId, MetaModels models) {
         ValidationResult validationResult = getValidationResult(archetypeId);
-        if(validationResult != null) {
+        Archetype archetype = getArchetype(archetypeId);
```

Comment:
Maybe you can do this to always get the validation results of the latest archetype without checking later?
```java
        Archetype archetype = getArchetype(archetypeId);
        ValidationResult validationResult = getValidationResult(archetype.getArchetypeId().getFullId());
```

Reply:
thanks, that sounds like a better suggestion indeed! I'll change it when adding a test :)